main.sh

```diff
@@ -30,6 +30,53 @@ warn() {
 info() {
     echo "info: $*"
 }
+cargo_binstall() {
+    tool="$1"
+    version="$2"
+
+    target="$(rustc -vV | grep host | cut -c 7-)"
+    base_url=https://github.com/ryankurte/cargo-binstall/releases/latest/download/cargo-binstall
+    is_zip=false
+    case "${target}" in
+        x86_64-unknown-linux-gnu) url="${base_url}-x86_64-unknown-linux-musl.tgz" ;;
+        x86_64-unknown-linux-musl) url="${base_url}-x86_64-unknown-linux-musl.tgz" ;;
+
+        armv7-unknown-linux-gnueabihf) url="${base_url}-armv7-unknown-linux-musleabihf.tgz" ;;
+        armv7-unknown-linux-musleabihf) url="${base_url}-armv7-unknown-linux-musleabihf.tgz" ;;
+
+        aarch64-unknown-linux-gnu) url="${base_url}-aarch64-unknown-linux-musl.tgz" ;;
+        aarch64-unknown-linux-musl) url="${base_url}-aarch64-unknown-linux-musl.tgz" ;;
+
+        x86_64-apple-darwin | aarch64-apple-darwin | x86_64-pc-windows-msvc)
+            is_zip=true
+            url="${base_url}-${target}.zip"
+            ;;
+
+        *) bail "unsupported target '${target}' for cargo-binstall" ;;
+    esac
+
+    cargo_bin="${CARGO_HOME:-~/.cargo}/bin"
+
+    if [ ! -f "${cargo_bin}/cargo-binstall" ]; then
+        if [ $is_zip = true ]; then
+            retry curl --proto '=https' --tlsv1.2 -fsSL --retry 10 --retry-connrefused "$url" -o "cargo-binstall-${target}.zip"
+            unzip "cargo-binstall-${target}.zip"
+            rm "cargo-binstall-${target}.zip"
+        else
+            retry curl --proto '=https' --tlsv1.2 -fsSL --retry 10 --retry-connrefused "$url" | tar xz
+        fi
+        mv cargo-binstall "${cargo_bin}"
+    fi
+
+    case "${version}" in
+        latest)
+            cargo binstall --no-confirm --target "$target" "$tool"
```

Comment:
There is a bug in `cargo-binstall` 0.8.0 that is going to be fixed here https://github.com/ryankurte/cargo-binstall/pull/157 and probably included in 0.8.1

Reply:
I will remove this once a new release of `cargo-binstall` that fixed this issue is published

Reply:
https://github.com/ryankurte/cargo-binstall/pull/161 and https://github.com/ryankurte/cargo-binstall/pull/157 shall fix this.

Reply:
[`cargo-binstall` v0.9.1](https://github.com/ryankurte/cargo-binstall/releases/tag/v0.9.1) has released and it not only fixed this problem, but also implement fallbacks:
 - on linux, it would first check for gnu (if present), then fallback to musl
 - on mac, it would first check for aarch64 (if on M1/M2), then fallback to x86_64

Reply:
Done!